daily_milestone.go

```diff
@@ -1,29 +1,221 @@
+	ProjectID   int        `json:"project_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+	State       string     `json:"state"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	CreatedAt   *time.Time `json:"created_at"`
+	Name        string     `json:"name"`
+	NameSpace   struct {
+		ID       int    `json:"id"`
+		Name     string `json:"name"`
+		Path     string `json:"path"`
+		Kind     string `json:"kind"`
+		FullPath string `json:"full_path"`
+	} `json:"namespace"`
+}
 
+// Initialization of logging variable
+var logger *log.Logger
+
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, token string, projectname string, namespace string) (string, error) {
+	projects := []gitLabAPI{}
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", baseURL, nil)
+	if err != nil {
+		logger.Println(err)
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
 	}
 
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+
+	}
+	json.Unmarshal(respByte, &projects)
+	defer resp.Body.Close()
+	for _, p := range projects {
+		// To check for message404 error and that project is not forund
+		if p.Name == "message" {
+			fmt.Println(p.Name)
+		}
+
+		if p.Name == projectname && p.NameSpace.Path == namespace {
+			fmt.Println(strconv.Itoa(p.ID))
+			return strconv.Itoa(p.ID), nil
+		}
+	}
+	return "", fmt.Errorf("project %s not found", projectname)
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// It is getting milestones data from the milestone API and returning list of closed milestones
+func getMilestones(baseURL string, token string, projectID string) ([]string, error) {
+	milestones := []milestoneAPI{}
+	list := []string{}
+	strurl := []string{baseURL, projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logger.Println(err)
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+	}
+
+	json.Unmarshal(respByte, &milestones)
+	defer resp.Body.Close()
+	for _, m := range milestones {
+		if m.State != "closed" {
+			titleTuple := []string{"Title: ", m.Title, ", DueDate", m.DueDate}
+			y := strings.Join(titleTuple, " ")
+			list = append(list, y)
```

Comment:
Any reason why this is named `y` ? 
If you can't easily come up with a name for a variable, sometimes it's because you don't even need the variable. Here you could just do `list = append(list, strings.Join(titleTuple, " "))`

Reply:
that's actually pretty good. will do it. Thanks